src/peripheral/dwt.rs

```diff
@@ -400,25 +411,50 @@ impl Comparator {
                     (_, EmitOption::PC) => return Err(DwtError::InvalidFunction),
                 };
 
-                self.function.modify(|mut r| {
-                    r.set_function(function);
-                    r.set_emitrange(emit_range);
-
+                (
+                    function,
+                    emit_range,
                     // don't compare data value
-                    r.set_datavmatch(false);
-
+                    false,
                     // don't compare cycle counter value
                     // NOTE: only needed for comparator 0, but is SBZP.
-                    r.set_cycmatch(false);
+                    false,
+                    settings.address,
+                    settings.mask,
+                )
+            }
+            ComparatorFunction::CycleCount(settings) => {
+                (
+                    // emit a Debug Watchpoint event, either halting execution or
+                    // firing a `DebugMonitor` exception
+                    0b0111,
```

Comment:
Please comment where this function is found in reference documentation.

Reply:
Done. Turns out I was looking at the wrong table and this value was technically unpredictable but worked for my chip! changed to the right value and referred to the table.